Makefile

```diff
@@ -30,7 +30,7 @@ docs: ## generate Sphinx HTML documentation, including API docs
 	$(BROWSER)docs/_build/html/index.html
 
 # Define PIP_COMPILE_OPTS=-v to get more information during make upgrade.
-PIP_COMPILE = pip-compile --rebuild --upgrade $(PIP_COMPILE_OPTS)
+PIP_COMPILE = pip-compile --rebuild --upgrade-package $(package)
```

Comment:
I'm surprised this doesn't cause an error when running `make upgrade`, since it would be getting an empty value.

Reply:
Oh, heh, actually it does! A different error was masking it.

OK, so I think what you want to do here is keep the previous options intact, but switch this to use conditional assignment (`?=`) so that it can be overridden by an environment variable.